Mysql-reactive client is not working on native mode

I am trying to run mysql-reactive client in native mode and it is not working while it works fine in JVM mode.

I am receiving the below exception when I try to connect to it :
```
java.util.concurrent.CompletionException: java.security.NoSuchAlgorithmException: RSA KeyFactory not available
bookrx_1  | 	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:164)
bookrx_1  | 	at org.jboss.resteasy.core.MethodInjectorImpl.lambda$invoke$3(MethodInjectorImpl.java:122)
bookrx_1  | 	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:616)
bookrx_1  | 	at java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:628)
bookrx_1  | 	at java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:1996)
bookrx_1  | 	at java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:110)
bookrx_1  | 	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:122)
bookrx_1  | 	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:594)
bookrx_1  | 	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:468)
bookrx_1  | 	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:421)
bookrx_1  | 	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:363)
bookrx_1  | 	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:423)
bookrx_1  | 	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:391)
bookrx_1  | 	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invoke$1(ResourceMethodInvoker.java:365)
bookrx_1  | 	at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:995)
bookrx_1  | 	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2137)
bookrx_1  | 	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:110)
bookrx_1  | 	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:365)
bookrx_1  | 	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:477)
bookrx_1  | 	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:252)
bookrx_1  | 	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:153)
bookrx_1  | 	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:363)
bookrx_1  | 	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:156)
bookrx_1  | 	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:238)
bookrx_1  | 	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
bookrx_1  | 	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:120)
bookrx_1  | 	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.access$000(VertxRequestHandler.java:36)
bookrx_1  | 	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:85)
bookrx_1  | 	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
bookrx_1  | 	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
bookrx_1  | 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
bookrx_1  | 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1395)
bookrx_1  | 	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
bookrx_1  | 	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
bookrx_1  | 	at java.lang.Thread.run(Thread.java:748)
bookrx_1  | 	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
bookrx_1  | 	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:460)
bookrx_1  | 	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
bookrx_1  | Caused by: java.util.concurrent.CompletionException: java.security.NoSuchAlgorithmException: RSA KeyFactory not available
bookrx_1  | 	at java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:375)
bookrx_1  | 	at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1947)
bookrx_1  | 	at org.ibra.ebs.service.BookService.init(BookService.java:38)
bookrx_1  | 	at org.ibra.ebs.service.BookService_Bean.create(BookService_Bean.zig:149)
bookrx_1  | 	at org.ibra.ebs.service.BookService_Bean.create(BookService_Bean.zig:343)
bookrx_1  | 	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:80)
bookrx_1  | 	at io.quarkus.arc.impl.ComputingCache$CacheFunction.lambda$apply$0(ComputingCache.java:99)
bookrx_1  | 	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
bookrx_1  | 	at io.quarkus.arc.impl.ComputingCache.getValue(ComputingCache.java:41)
bookrx_1  | 	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:25)
bookrx_1  | 	at org.ibra.ebs.service.BookService_ClientProxy.arc$delegate(BookService_ClientProxy.zig:99)
bookrx_1  | 	at org.ibra.ebs.service.BookService_ClientProxy.getLastestRecords(BookService_ClientProxy.zig:248)
bookrx_1  | 	at org.ibra.ebs.resource.BookResource.get(BookResource.java:47)
bookrx_1  | 	at org.ibra.ebs.resource.BookResource_ClientProxy.get(BookResource_ClientProxy.zig:252)
bookrx_1  | 	at java.lang.reflect.Method.invoke(Method.java:498)
bookrx_1  | 	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:151)
bookrx_1  | 	... 37 more
bookrx_1  | Caused by: java.security.NoSuchAlgorithmException: RSA KeyFactory not available
bookrx_1  | 	at java.security.KeyFactory.<init>(KeyFactory.java:138)
bookrx_1  | 	at java.security.KeyFactory.getInstance(KeyFactory.java:172)
bookrx_1  | 	at io.vertx.mysqlclient.impl.util.RsaPublicKeyEncryptor.generateRsaPublicKey(RsaPublicKeyEncryptor.java:34)
bookrx_1  | 	at io.vertx.mysqlclient.impl.util.RsaPublicKeyEncryptor.encrypt(RsaPublicKeyEncryptor.java:21)
bookrx_1  | 	at io.vertx.mysqlclient.impl.codec.AuthenticationCommandBaseCodec.sendEncryptedPasswordWithServerRsaPublicKey(AuthenticationCommandBaseCodec.java:76)
bookrx_1  | 	at io.vertx.mysqlclient.impl.codec.AuthenticationCommandBaseCodec.handleAuthMoreData(AuthenticationCommandBaseCodec.java:35)
bookrx_1  | 	at io.vertx.mysqlclient.impl.codec.InitialHandshakeCommandCodec.handleAuthentication(InitialHandshakeCommandCodec.java:188)
bookrx_1  | 	at io.vertx.mysqlclient.impl.codec.InitialHandshakeCommandCodec.decodePayload(InitialHandshakeCommandCodec.java:60)
bookrx_1  | 	at io.vertx.mysqlclient.impl.codec.MySQLDecoder.decodePayload(MySQLDecoder.java:62)
bookrx_1  | 	at io.vertx.mysqlclient.impl.codec.MySQLDecoder.decode(MySQLDecoder.java:51)
bookrx_1  | 	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:505)
bookrx_1  | 	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:444)
bookrx_1  | 	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:283)
bookrx_1  | 	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:253)
bookrx_1  | 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
bookrx_1  | 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
bookrx_1  | 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352)
bookrx_1  | 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1422)
bookrx_1  | 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
bookrx_1  | 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
bookrx_1  | 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:931)
bookrx_1  | 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
bookrx_1  | 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:700)
bookrx_1  | 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:635)
bookrx_1  | 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:552)
bookrx_1  | 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:514)
bookrx_1  | 	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044)
bookrx_1  | 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
bookrx_1  | 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
bookrx_1  | 	at java.lang.Thread.run(Thread.java:748)
bookrx_1  | 	... 2 more
```

**Configuration**
```properties
quarkus.datasource.url = vertx-reactive:mysql://myhost:3306/ebs_book
quarkus.datasource.username = xxx
quarkus.datasource.password = xxx
```

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
18.7.0 Darwin Kernel Version 18.7.0: Sat Oct 12 00:02:19 PDT 2019; root:xnu-4903.278.12~1/RELEASE_X86_64 x86_64
 - Output of `java -version`: 
openjdk version "1.8.0_202"
OpenJDK Runtime Environment Corretto-8.202.08.2 (build 1.8.0_202-b08)
OpenJDK 64-Bit Server VM Corretto-8.202.08.2 (build 25.202-b08, mixed mode)
 - GraalVM version (if different from Java): 
graalvm-ce-19.0.2
 - Quarkus version or git rev: 
1.1.0.Final

Also tried this [enabling SSL for my image](https://quarkus.io/guides/native-and-ssl#working-with-containers) and it didn't work.

--- a/docs/src/main/asciidoc/native-and-ssl.adoc
+++ b/docs/src/main/asciidoc/native-and-ssl.adoc
@@ -79,7 +79,7 @@ As SSL is de facto the standard nowadays, we decided to enable its support autom
  * the OAuth2 extension (`quarkus-elytron-security-oauth2`),
  * the REST client extension (`quarkus-rest-client`),
  * the Reactive client for PostgreSQL extension (`quarkus-reactive-pg-client`).
-
+ * the Reactive client for MySQL extension (`quarkus-reactive-mysql-client`).
 
 As long as you have one of those extensions in your project, the SSL support will be enabled by default.
 

--- a/docs/src/main/asciidoc/native-and-ssl.adoc
+++ b/docs/src/main/asciidoc/native-and-ssl.adoc
@@ -77,7 +77,8 @@ As SSL is de facto the standard nowadays, we decided to enable its support autom
  * the MongoDB extension (`quarkus-mongodb-client`),
  * the Neo4j extension (`quarkus-neo4j`),
  * the OAuth2 extension (`quarkus-elytron-security-oauth2`),
- * the REST client extension (`quarkus-rest-client`).
+ * the REST client extension (`quarkus-rest-client`),
+ * the Reactive client for PostgreSQL extension (`quarkus-reactive-pg-client`).
 
 
 As long as you have one of those extensions in your project, the SSL support will be enabled by default.

--- a/extensions/reactive-datasource/runtime/pom.xml
+++ b/extensions/reactive-datasource/runtime/pom.xml
@@ -30,6 +30,10 @@
             <groupId>io.vertx</groupId>
             <artifactId>vertx-sql-client</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-vertx-core</artifactId>
+        </dependency>
     </dependencies>
 
     <build>

--- a/extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java
+++ b/extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java
@@ -38,7 +38,7 @@ public class DataSourceReactiveRuntimeConfig {
     /**
      * Trust configuration in the PEM format.
      * <p>
-     * When enabled, {@link #trustCertificateJks} and {@link #trustCertificatePfx} must be disabled.
+     * When enabled, {@code #trust-certificate-jks} and {@code #trust-certificate-pfx} must be disabled.
      */
     @ConfigItem
     public PemTrustCertConfiguration trustCertificatePem;
@@ -46,7 +46,7 @@ public class DataSourceReactiveRuntimeConfig {
     /**
      * Trust configuration in the JKS format.
      * <p>
-     * When enabled, {@link #trustCertificatePem} and {@link #trustCertificatePfx} must be disabled.
+     * When enabled, {@code #trust-certificate-pem} and {@code #trust-certificate-pfx} must be disabled.
      */
     @ConfigItem
     public JksConfiguration trustCertificateJks;
@@ -54,7 +54,7 @@ public class DataSourceReactiveRuntimeConfig {
     /**
      * Trust configuration in the PFX format.
      * <p>
-     * When enabled, {@link #trustCertificateJks} and {@link #trustCertificatePem} must be disabled.
+     * When enabled, {@code #trust-certificate-jks} and {@code #trust-certificate-pem} must be disabled.
      */
     @ConfigItem
     public PfxConfiguration trustCertificatePfx;
@@ -62,7 +62,7 @@ public class DataSourceReactiveRuntimeConfig {
     /**
      * Key/cert configuration in the PEM format.
      * <p>
-     * When enabled, {@link #keyCertificateJks} and {@link #keyCertificatePfx} must be disabled.
+     * When enabled, {@code key-certificate-jks} and {@code #key-certificate-pfx} must be disabled.
      */
     @ConfigItem
     public PemKeyCertConfiguration keyCertificatePem;
@@ -70,7 +70,7 @@ public class DataSourceReactiveRuntimeConfig {
     /**
      * Key/cert configuration in the JKS format.
      * <p>
-     * When enabled, {@link #keyCertificatePem} and {@link #keyCertificatePfx} must be disabled.
+     * When enabled, {@code #key-certificate-pem} and {@code #key-certificate-pfx} must be disabled.
      */
     @ConfigItem
     public JksConfiguration keyCertificateJks;
@@ -78,7 +78,7 @@ public class DataSourceReactiveRuntimeConfig {
     /**
      * Key/cert configuration in the PFX format.
      * <p>
-     * When enabled, {@link #keyCertificateJks} and {@link #keyCertificatePem} must be disabled.
+     * When enabled, {@code key-certificate-jks} and {@code #key-certificate-pem} must be disabled.
      */
     @ConfigItem
     public PfxConfiguration keyCertificatePfx;

--- a/extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java
+++ b/extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java
@@ -6,6 +6,10 @@ import java.util.OptionalInt;
 import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
+import io.quarkus.vertx.core.runtime.config.JksConfiguration;
+import io.quarkus.vertx.core.runtime.config.PemKeyCertConfiguration;
+import io.quarkus.vertx.core.runtime.config.PemTrustCertConfiguration;
+import io.quarkus.vertx.core.runtime.config.PfxConfiguration;
 
 /**
  * For now, the reactive extensions only support a default datasource.
@@ -24,4 +28,58 @@ public class DataSourceReactiveRuntimeConfig {
      */
     @ConfigItem
     public OptionalInt maxSize;
+
+    /**
+     * Whether all server certificates should be trusted.
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean trustAll;
+
+    /**
+     * Trust configuration in the PEM format.
+     * <p>
+     * When enabled, {@link #trustCertificateJks} and {@link #trustCertificatePfx} must be disabled.
+     */
+    @ConfigItem
+    public PemTrustCertConfiguration trustCertificatePem;
+
+    /**
+     * Trust configuration in the JKS format.
+     * <p>
+     * When enabled, {@link #trustCertificatePem} and {@link #trustCertificatePfx} must be disabled.
+     */
+    @ConfigItem
+    public JksConfiguration trustCertificateJks;
+
+    /**
+     * Trust configuration in the PFX format.
+     * <p>
+     * When enabled, {@link #trustCertificateJks} and {@link #trustCertificatePem} must be disabled.
+     */
+    @ConfigItem
+    public PfxConfiguration trustCertificatePfx;
+
+    /**
+     * Key/cert configuration in the PEM format.
+     * <p>
+     * When enabled, {@link #keyCertificateJks} and {@link #keyCertificatePfx} must be disabled.
+     */
+    @ConfigItem
+    public PemKeyCertConfiguration keyCertificatePem;
+
+    /**
+     * Key/cert configuration in the JKS format.
+     * <p>
+     * When enabled, {@link #keyCertificatePem} and {@link #keyCertificatePfx} must be disabled.
+     */
+    @ConfigItem
+    public JksConfiguration keyCertificateJks;
+
+    /**
+     * Key/cert configuration in the PFX format.
+     * <p>
+     * When enabled, {@link #keyCertificateJks} and {@link #keyCertificatePem} must be disabled.
+     */
+    @ConfigItem
+    public PfxConfiguration keyCertificatePfx;
 }

--- a/extensions/reactive-mysql-client/deployment/src/main/java/io/quarkus/reactive/mysql/client/deployment/ReactiveMySQLClientProcessor.java
+++ b/extensions/reactive-mysql-client/deployment/src/main/java/io/quarkus/reactive/mysql/client/deployment/ReactiveMySQLClientProcessor.java
@@ -11,6 +11,7 @@ import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.ExecutionTime;
 import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
 import io.quarkus.deployment.builditem.ServiceStartBuildItem;
 import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
@@ -42,6 +43,7 @@ class ReactiveMySQLClientProcessor {
             MySQLPoolRecorder recorder,
             VertxBuildItem vertx,
             BeanContainerBuildItem beanContainer, ShutdownContextBuildItem shutdown,
+            BuildProducer<ExtensionSslNativeSupportBuildItem> sslNativeSupport,
             DataSourcesBuildTimeConfig dataSourcesBuildTimeConfig, DataSourcesRuntimeConfig dataSourcesRuntimeConfig,
             DataSourceReactiveBuildTimeConfig dataSourceReactiveBuildTimeConfig,
             DataSourceReactiveRuntimeConfig dataSourceReactiveRuntimeConfig,
@@ -71,6 +73,9 @@ class ReactiveMySQLClientProcessor {
 
         boolean isDefault = true; // assume always the default pool for now
         vertxPool.produce(new VertxPoolBuildItem(mySqlPool, DatabaseKind.MYSQL, isDefault));
+
+        // Enable SSL support by default
+        sslNativeSupport.produce(new ExtensionSslNativeSupportBuildItem(Feature.REACTIVE_MYSQL_CLIENT));
 
         return serviceStart;
     }

--- a/extensions/reactive-mysql-client/runtime/src/main/java/io/quarkus/reactive/mysql/client/runtime/DataSourceReactiveMySQLConfig.java
+++ b/extensions/reactive-mysql-client/runtime/src/main/java/io/quarkus/reactive/mysql/client/runtime/DataSourceReactiveMySQLConfig.java
@@ -5,6 +5,7 @@ import java.util.Optional;
 import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
+import io.vertx.mysqlclient.SslMode;
 
 @ConfigRoot(name = "datasource.reactive.mysql", phase = ConfigPhase.RUN_TIME)
 public class DataSourceReactiveMySQLConfig {
@@ -26,4 +27,13 @@ public class DataSourceReactiveMySQLConfig {
      */
     @ConfigItem
     public Optional<String> collation;
+
+    /**
+     * Desired security state of the connection to the server.
+     * <p>
+     * See <a href="https://dev.mysql.com/doc/refman/8.0/en/connection-options.html#option_general_ssl-mode">MySQL Reference
+     * Manual</a>.
+     */
+    @ConfigItem(defaultValueDocumentation = "disabled")
+    public Optional<SslMode> sslMode;
 }

--- a/extensions/reactive-mysql-client/runtime/src/main/java/io/quarkus/reactive/mysql/client/runtime/MySQLPoolRecorder.java
+++ b/extensions/reactive-mysql-client/runtime/src/main/java/io/quarkus/reactive/mysql/client/runtime/MySQLPoolRecorder.java
@@ -2,6 +2,12 @@ package io.quarkus.reactive.mysql.client.runtime;
 
 import static io.quarkus.credentials.CredentialsProvider.PASSWORD_PROPERTY_NAME;
 import static io.quarkus.credentials.CredentialsProvider.USER_PROPERTY_NAME;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configureJksKeyCertOptions;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configureJksTrustOptions;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configurePemKeyCertOptions;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configurePemTrustOptions;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configurePfxKeyCertOptions;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configurePfxTrustOptions;
 
 import java.util.Map;
 
@@ -121,6 +127,20 @@ public class MySQLPoolRecorder {
         if (dataSourceReactiveMySQLConfig.collation.isPresent()) {
             mysqlConnectOptions.setCollation(dataSourceReactiveMySQLConfig.collation.get());
         }
+
+        if (dataSourceReactiveMySQLConfig.sslMode.isPresent()) {
+            mysqlConnectOptions.setSslMode(dataSourceReactiveMySQLConfig.sslMode.get());
+        }
+
+        mysqlConnectOptions.setTrustAll(dataSourceReactiveRuntimeConfig.trustAll);
+
+        configurePemTrustOptions(mysqlConnectOptions, dataSourceReactiveRuntimeConfig.trustCertificatePem);
+        configureJksTrustOptions(mysqlConnectOptions, dataSourceReactiveRuntimeConfig.trustCertificateJks);
+        configurePfxTrustOptions(mysqlConnectOptions, dataSourceReactiveRuntimeConfig.trustCertificatePfx);
+
+        configurePemKeyCertOptions(mysqlConnectOptions, dataSourceReactiveRuntimeConfig.keyCertificatePem);
+        configureJksKeyCertOptions(mysqlConnectOptions, dataSourceReactiveRuntimeConfig.keyCertificateJks);
+        configurePfxKeyCertOptions(mysqlConnectOptions, dataSourceReactiveRuntimeConfig.keyCertificatePfx);
 
         return mysqlConnectOptions;
     }

--- a/extensions/reactive-pg-client/deployment/src/main/java/io/quarkus/reactive/pg/client/deployment/ReactivePgClientProcessor.java
+++ b/extensions/reactive-pg-client/deployment/src/main/java/io/quarkus/reactive/pg/client/deployment/ReactivePgClientProcessor.java
@@ -11,6 +11,7 @@ import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.ExecutionTime;
 import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
 import io.quarkus.deployment.builditem.ServiceStartBuildItem;
 import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
@@ -50,6 +51,7 @@ class ReactivePgClientProcessor {
             VertxBuildItem vertx,
             BeanContainerBuildItem beanContainer,
             ShutdownContextBuildItem shutdown,
+            BuildProducer<ExtensionSslNativeSupportBuildItem> sslNativeSupport,
             DataSourcesBuildTimeConfig dataSourcesBuildTimeConfig, DataSourcesRuntimeConfig dataSourcesRuntimeConfig,
             DataSourceReactiveBuildTimeConfig dataSourceReactiveBuildTimeConfig,
             DataSourceReactiveRuntimeConfig dataSourceReactiveRuntimeConfig,
@@ -78,6 +80,9 @@ class ReactivePgClientProcessor {
 
         boolean isDefault = true; // assume always the default pool for now
         vertxPool.produce(new VertxPoolBuildItem(pool, DatabaseKind.POSTGRESQL, isDefault));
+
+        // Enable SSL support by default
+        sslNativeSupport.produce(new ExtensionSslNativeSupportBuildItem(Feature.REACTIVE_PG_CLIENT));
 
         return serviceStart;
     }

--- a/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/DataSourceReactivePostgreSQLConfig.java
+++ b/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/DataSourceReactivePostgreSQLConfig.java
@@ -40,35 +40,35 @@ public class DataSourceReactivePostgreSQLConfig {
      * Trust configuration in the PEM format.
      */
     @ConfigItem
-    public PemTrustCertConfiguration trustPem;
+    public PemTrustCertConfiguration trustCertificatePem;
 
     /**
      * Trust configuration in the JKS format.
      */
     @ConfigItem
-    public JksConfiguration trustJks;
+    public JksConfiguration trustCertificateJks;
 
     /**
      * Trust configuration in the PFX format.
      */
     @ConfigItem
-    public PfxConfiguration trustPfx;
+    public PfxConfiguration trustCertificatePfx;
 
     /**
      * Key/cert configuration in the PEM format.
      */
     @ConfigItem
-    public PemKeyCertConfiguration keyCertPem;
+    public PemKeyCertConfiguration keyCertificatePem;
 
     /**
      * Key/cert configuration in the JKS format.
      */
     @ConfigItem
-    public JksConfiguration keyCertJks;
+    public JksConfiguration keyCertificateJks;
 
     /**
      * Key/cert configuration in the PFX format.
      */
     @ConfigItem
-    public PfxConfiguration keyCertPfx;
+    public PfxConfiguration keyCertificatePfx;
 }

--- a/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/DataSourceReactivePostgreSQLConfig.java
+++ b/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/DataSourceReactivePostgreSQLConfig.java
@@ -6,6 +6,11 @@ import java.util.OptionalInt;
 import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
+import io.quarkus.vertx.core.runtime.config.JksConfiguration;
+import io.quarkus.vertx.core.runtime.config.PemKeyCertConfiguration;
+import io.quarkus.vertx.core.runtime.config.PemTrustCertConfiguration;
+import io.quarkus.vertx.core.runtime.config.PfxConfiguration;
+import io.vertx.pgclient.SslMode;
 
 @ConfigRoot(name = "datasource.reactive.postgresql", phase = ConfigPhase.RUN_TIME)
 public class DataSourceReactivePostgreSQLConfig {
@@ -21,4 +26,49 @@ public class DataSourceReactivePostgreSQLConfig {
      */
     @ConfigItem
     public OptionalInt pipeliningLimit;
+
+    /**
+     * SSL operating mode of the client.
+     * <p>
+     * See <a href="https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-PROTECTION">Protection Provided in
+     * Different Modes</a>.
+     */
+    @ConfigItem
+    public Optional<SslMode> sslMode;
+
+    /**
+     * Trust configuration in the PEM format.
+     */
+    @ConfigItem
+    public PemTrustCertConfiguration trustPem;
+
+    /**
+     * Trust configuration in the JKS format.
+     */
+    @ConfigItem
+    public JksConfiguration trustJks;
+
+    /**
+     * Trust configuration in the PFX format.
+     */
+    @ConfigItem
+    public PfxConfiguration trustPfx;
+
+    /**
+     * Key/cert configuration in the PEM format.
+     */
+    @ConfigItem
+    public PemKeyCertConfiguration keyCertPem;
+
+    /**
+     * Key/cert configuration in the JKS format.
+     */
+    @ConfigItem
+    public JksConfiguration keyCertJks;
+
+    /**
+     * Key/cert configuration in the PFX format.
+     */
+    @ConfigItem
+    public PfxConfiguration keyCertPfx;
 }

--- a/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/DataSourceReactivePostgreSQLConfig.java
+++ b/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/DataSourceReactivePostgreSQLConfig.java
@@ -33,41 +33,53 @@ public class DataSourceReactivePostgreSQLConfig {
      * See <a href="https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-PROTECTION">Protection Provided in
      * Different Modes</a>.
      */
-    @ConfigItem
+    @ConfigItem(defaultValueDocumentation = "disable")
     public Optional<SslMode> sslMode;
 
     /**
      * Trust configuration in the PEM format.
+     * <p>
+     * When enabled, {@link #trustCertificateJks} and {@link #trustCertificatePfx} must be disabled.
      */
     @ConfigItem
     public PemTrustCertConfiguration trustCertificatePem;
 
     /**
      * Trust configuration in the JKS format.
+     * <p>
+     * When enabled, {@link #trustCertificatePem} and {@link #trustCertificatePfx} must be disabled.
      */
     @ConfigItem
     public JksConfiguration trustCertificateJks;
 
     /**
      * Trust configuration in the PFX format.
+     * <p>
+     * When enabled, {@link #trustCertificateJks} and {@link #trustCertificatePem} must be disabled.
      */
     @ConfigItem
     public PfxConfiguration trustCertificatePfx;
 
     /**
      * Key/cert configuration in the PEM format.
+     * <p>
+     * When enabled, {@link #keyCertificateJks} and {@link #keyCertificatePfx} must be disabled.
      */
     @ConfigItem
     public PemKeyCertConfiguration keyCertificatePem;
 
     /**
      * Key/cert configuration in the JKS format.
+     * <p>
+     * When enabled, {@link #keyCertificatePem} and {@link #keyCertificatePfx} must be disabled.
      */
     @ConfigItem
     public JksConfiguration keyCertificateJks;
 
     /**
      * Key/cert configuration in the PFX format.
+     * <p>
+     * When enabled, {@link #keyCertificateJks} and {@link #keyCertificatePem} must be disabled.
      */
     @ConfigItem
     public PfxConfiguration keyCertificatePfx;

--- a/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/DataSourceReactivePostgreSQLConfig.java
+++ b/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/DataSourceReactivePostgreSQLConfig.java
@@ -6,10 +6,6 @@ import java.util.OptionalInt;
 import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
-import io.quarkus.vertx.core.runtime.config.JksConfiguration;
-import io.quarkus.vertx.core.runtime.config.PemKeyCertConfiguration;
-import io.quarkus.vertx.core.runtime.config.PemTrustCertConfiguration;
-import io.quarkus.vertx.core.runtime.config.PfxConfiguration;
 import io.vertx.pgclient.SslMode;
 
 @ConfigRoot(name = "datasource.reactive.postgresql", phase = ConfigPhase.RUN_TIME)
@@ -35,52 +31,4 @@ public class DataSourceReactivePostgreSQLConfig {
      */
     @ConfigItem(defaultValueDocumentation = "disable")
     public Optional<SslMode> sslMode;
-
-    /**
-     * Trust configuration in the PEM format.
-     * <p>
-     * When enabled, {@link #trustCertificateJks} and {@link #trustCertificatePfx} must be disabled.
-     */
-    @ConfigItem
-    public PemTrustCertConfiguration trustCertificatePem;
-
-    /**
-     * Trust configuration in the JKS format.
-     * <p>
-     * When enabled, {@link #trustCertificatePem} and {@link #trustCertificatePfx} must be disabled.
-     */
-    @ConfigItem
-    public JksConfiguration trustCertificateJks;
-
-    /**
-     * Trust configuration in the PFX format.
-     * <p>
-     * When enabled, {@link #trustCertificateJks} and {@link #trustCertificatePem} must be disabled.
-     */
-    @ConfigItem
-    public PfxConfiguration trustCertificatePfx;
-
-    /**
-     * Key/cert configuration in the PEM format.
-     * <p>
-     * When enabled, {@link #keyCertificateJks} and {@link #keyCertificatePfx} must be disabled.
-     */
-    @ConfigItem
-    public PemKeyCertConfiguration keyCertificatePem;
-
-    /**
-     * Key/cert configuration in the JKS format.
-     * <p>
-     * When enabled, {@link #keyCertificatePem} and {@link #keyCertificatePfx} must be disabled.
-     */
-    @ConfigItem
-    public JksConfiguration keyCertificateJks;
-
-    /**
-     * Key/cert configuration in the PFX format.
-     * <p>
-     * When enabled, {@link #keyCertificateJks} and {@link #keyCertificatePem} must be disabled.
-     */
-    @ConfigItem
-    public PfxConfiguration keyCertificatePfx;
 }

--- a/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolRecorder.java
+++ b/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolRecorder.java
@@ -126,13 +126,13 @@ public class PgPoolRecorder {
             pgConnectOptions.setSslMode(dataSourceReactivePostgreSQLConfig.sslMode.get());
         }
 
-        configurePemTrustOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.trustPem);
-        configureJksTrustOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.trustJks);
-        configurePfxTrustOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.trustPfx);
+        configurePemTrustOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.trustCertificatePem);
+        configureJksTrustOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.trustCertificateJks);
+        configurePfxTrustOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.trustCertificatePfx);
 
-        configurePemKeyCertOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.keyCertPem);
-        configureJksKeyCertOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.keyCertJks);
-        configurePfxKeyCertOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.keyCertPfx);
+        configurePemKeyCertOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.keyCertificatePem);
+        configureJksKeyCertOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.keyCertificateJks);
+        configurePfxKeyCertOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.keyCertificatePfx);
 
         return pgConnectOptions;
     }

--- a/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolRecorder.java
+++ b/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolRecorder.java
@@ -2,7 +2,12 @@ package io.quarkus.reactive.pg.client.runtime;
 
 import static io.quarkus.credentials.CredentialsProvider.PASSWORD_PROPERTY_NAME;
 import static io.quarkus.credentials.CredentialsProvider.USER_PROPERTY_NAME;
-import static io.quarkus.vertx.core.runtime.SSLConfigHelper.*;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configureJksKeyCertOptions;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configureJksTrustOptions;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configurePemKeyCertOptions;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configurePemTrustOptions;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configurePfxKeyCertOptions;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configurePfxTrustOptions;
 
 import java.util.Map;
 

--- a/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolRecorder.java
+++ b/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolRecorder.java
@@ -131,13 +131,15 @@ public class PgPoolRecorder {
             pgConnectOptions.setSslMode(dataSourceReactivePostgreSQLConfig.sslMode.get());
         }
 
-        configurePemTrustOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.trustCertificatePem);
-        configureJksTrustOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.trustCertificateJks);
-        configurePfxTrustOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.trustCertificatePfx);
+        pgConnectOptions.setTrustAll(dataSourceReactiveRuntimeConfig.trustAll);
 
-        configurePemKeyCertOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.keyCertificatePem);
-        configureJksKeyCertOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.keyCertificateJks);
-        configurePfxKeyCertOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.keyCertificatePfx);
+        configurePemTrustOptions(pgConnectOptions, dataSourceReactiveRuntimeConfig.trustCertificatePem);
+        configureJksTrustOptions(pgConnectOptions, dataSourceReactiveRuntimeConfig.trustCertificateJks);
+        configurePfxTrustOptions(pgConnectOptions, dataSourceReactiveRuntimeConfig.trustCertificatePfx);
+
+        configurePemKeyCertOptions(pgConnectOptions, dataSourceReactiveRuntimeConfig.keyCertificatePem);
+        configureJksKeyCertOptions(pgConnectOptions, dataSourceReactiveRuntimeConfig.keyCertificateJks);
+        configurePfxKeyCertOptions(pgConnectOptions, dataSourceReactiveRuntimeConfig.keyCertificatePfx);
 
         return pgConnectOptions;
     }

--- a/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolRecorder.java
+++ b/extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolRecorder.java
@@ -2,6 +2,7 @@ package io.quarkus.reactive.pg.client.runtime;
 
 import static io.quarkus.credentials.CredentialsProvider.PASSWORD_PROPERTY_NAME;
 import static io.quarkus.credentials.CredentialsProvider.USER_PROPERTY_NAME;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.*;
 
 import java.util.Map;
 
@@ -120,6 +121,18 @@ public class PgPoolRecorder {
         if (dataSourceReactivePostgreSQLConfig.pipeliningLimit.isPresent()) {
             pgConnectOptions.setPipeliningLimit(dataSourceReactivePostgreSQLConfig.pipeliningLimit.getAsInt());
         }
+
+        if (dataSourceReactivePostgreSQLConfig.sslMode.isPresent()) {
+            pgConnectOptions.setSslMode(dataSourceReactivePostgreSQLConfig.sslMode.get());
+        }
+
+        configurePemTrustOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.trustPem);
+        configureJksTrustOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.trustJks);
+        configurePfxTrustOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.trustPfx);
+
+        configurePemKeyCertOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.keyCertPem);
+        configureJksKeyCertOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.keyCertJks);
+        configurePfxKeyCertOptions(pgConnectOptions, dataSourceReactivePostgreSQLConfig.keyCertPfx);
 
         return pgConnectOptions;
     }

--- a/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/SSLConfigHelper.java
+++ b/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/SSLConfigHelper.java
@@ -1,16 +1,17 @@
 package io.quarkus.vertx.core.runtime;
 
-import java.util.regex.Pattern;
-
 import io.quarkus.vertx.core.runtime.config.JksConfiguration;
 import io.quarkus.vertx.core.runtime.config.PemKeyCertConfiguration;
 import io.quarkus.vertx.core.runtime.config.PemTrustCertConfiguration;
 import io.quarkus.vertx.core.runtime.config.PfxConfiguration;
-import io.vertx.core.net.*;
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.KeyCertOptions;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.PemTrustOptions;
+import io.vertx.core.net.PfxOptions;
+import io.vertx.core.net.TCPSSLOptions;
 
 public class SSLConfigHelper {
-
-    private static final Pattern COMMA_PATTERN = Pattern.compile(",");
 
     public static void configurePemTrustOptions(TCPSSLOptions options, PemTrustCertConfiguration configuration) {
         if (configuration.enabled) {
@@ -22,8 +23,8 @@ public class SSLConfigHelper {
     private static PemTrustOptions toPemTrustOptions(PemTrustCertConfiguration configuration) {
         PemTrustOptions pemTrustOptions = new PemTrustOptions();
         if (configuration.certs.isPresent()) {
-            for (String cert : COMMA_PATTERN.split(configuration.certs.get())) {
-                pemTrustOptions.addCertPath(cert.trim());
+            for (String cert : configuration.certs.get()) {
+                pemTrustOptions.addCertPath(cert);
             }
         }
         return pemTrustOptions;
@@ -81,13 +82,13 @@ public class SSLConfigHelper {
     private static KeyCertOptions toPemKeyCertOptions(PemKeyCertConfiguration configuration) {
         PemKeyCertOptions pemKeyCertOptions = new PemKeyCertOptions();
         if (configuration.certs.isPresent()) {
-            for (String cert : COMMA_PATTERN.split(configuration.certs.get())) {
-                pemKeyCertOptions.addCertPath(cert.trim());
+            for (String cert : configuration.certs.get()) {
+                pemKeyCertOptions.addCertPath(cert);
             }
         }
         if (configuration.keys.isPresent()) {
-            for (String cert : COMMA_PATTERN.split(configuration.keys.get())) {
-                pemKeyCertOptions.addKeyPath(cert.trim());
+            for (String cert : configuration.keys.get()) {
+                pemKeyCertOptions.addKeyPath(cert);
             }
         }
         return pemKeyCertOptions;

--- a/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/SSLConfigHelper.java
+++ b/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/SSLConfigHelper.java
@@ -1,0 +1,119 @@
+package io.quarkus.vertx.core.runtime;
+
+import java.util.regex.Pattern;
+
+import io.quarkus.vertx.core.runtime.config.JksConfiguration;
+import io.quarkus.vertx.core.runtime.config.PemKeyCertConfiguration;
+import io.quarkus.vertx.core.runtime.config.PemTrustCertConfiguration;
+import io.quarkus.vertx.core.runtime.config.PfxConfiguration;
+import io.vertx.core.net.*;
+
+public class SSLConfigHelper {
+
+    private static final Pattern COMMA_PATTERN = Pattern.compile(",");
+
+    public static void configurePemTrustOptions(TCPSSLOptions options, PemTrustCertConfiguration configuration) {
+        if (configuration.enabled) {
+            ensureTrustOptionsNotSet(options);
+            options.setTrustOptions(toPemTrustOptions(configuration));
+        }
+    }
+
+    private static PemTrustOptions toPemTrustOptions(PemTrustCertConfiguration configuration) {
+        PemTrustOptions pemTrustOptions = new PemTrustOptions();
+        if (configuration.certs.isPresent()) {
+            for (String cert : COMMA_PATTERN.split(configuration.certs.get())) {
+                pemTrustOptions.addCertPath(cert.trim());
+            }
+        }
+        return pemTrustOptions;
+    }
+
+    public static void configureJksTrustOptions(TCPSSLOptions options, JksConfiguration configuration) {
+        if (configuration.enabled) {
+            ensureTrustOptionsNotSet(options);
+            options.setTrustOptions(toJksOptions(configuration));
+        }
+    }
+
+    private static JksOptions toJksOptions(JksConfiguration configuration) {
+        JksOptions jksOptions = new JksOptions();
+        if (configuration.path.isPresent()) {
+            jksOptions.setPath(configuration.path.get());
+        }
+        if (configuration.password.isPresent()) {
+            jksOptions.setPassword(configuration.password.get());
+        }
+        return jksOptions;
+    }
+
+    public static void configurePfxTrustOptions(TCPSSLOptions options, PfxConfiguration configuration) {
+        if (configuration.enabled) {
+            ensureTrustOptionsNotSet(options);
+            options.setTrustOptions(toPfxOptions(configuration));
+        }
+    }
+
+    private static PfxOptions toPfxOptions(PfxConfiguration configuration) {
+        PfxOptions pfxOptions = new PfxOptions();
+        if (configuration.path.isPresent()) {
+            pfxOptions.setPath(configuration.path.get());
+        }
+        if (configuration.password.isPresent()) {
+            pfxOptions.setPassword(configuration.password.get());
+        }
+        return pfxOptions;
+    }
+
+    private static void ensureTrustOptionsNotSet(TCPSSLOptions options) {
+        if (options.getTrustOptions() != null) {
+            throw new IllegalArgumentException("Trust options have already been set");
+        }
+    }
+
+    public static void configurePemKeyCertOptions(TCPSSLOptions options, PemKeyCertConfiguration configuration) {
+        if (configuration.enabled) {
+            ensureKeyCertOptionsNotSet(options);
+            options.setKeyCertOptions(toPemKeyCertOptions(configuration));
+        }
+    }
+
+    private static KeyCertOptions toPemKeyCertOptions(PemKeyCertConfiguration configuration) {
+        PemKeyCertOptions pemKeyCertOptions = new PemKeyCertOptions();
+        if (configuration.certs.isPresent()) {
+            for (String cert : COMMA_PATTERN.split(configuration.certs.get())) {
+                pemKeyCertOptions.addCertPath(cert.trim());
+            }
+        }
+        if (configuration.keys.isPresent()) {
+            for (String cert : COMMA_PATTERN.split(configuration.keys.get())) {
+                pemKeyCertOptions.addKeyPath(cert.trim());
+            }
+        }
+        return pemKeyCertOptions;
+    }
+
+    public static void configureJksKeyCertOptions(TCPSSLOptions options, JksConfiguration configuration) {
+        if (configuration.enabled) {
+            ensureKeyCertOptionsNotSet(options);
+            options.setKeyCertOptions(toJksOptions(configuration));
+        }
+    }
+
+    public static void configurePfxKeyCertOptions(TCPSSLOptions options, PfxConfiguration configuration) {
+        if (configuration.enabled) {
+            ensureKeyCertOptionsNotSet(options);
+            options.setKeyCertOptions(toPfxOptions(configuration));
+        }
+    }
+
+    private static void ensureKeyCertOptionsNotSet(TCPSSLOptions options) {
+        if (options.getKeyCertOptions() != null) {
+            throw new IllegalArgumentException("Key cert options have already been set");
+        }
+    }
+
+    private SSLConfigHelper() {
+        // Utility
+    }
+}

--- a/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java
+++ b/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java
@@ -1,10 +1,14 @@
 package io.quarkus.vertx.core.runtime;
 
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configureJksKeyCertOptions;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configurePemKeyCertOptions;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configurePemTrustOptions;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configurePfxKeyCertOptions;
+import static io.quarkus.vertx.core.runtime.SSLConfigHelper.configurePfxTrustOptions;
 import static io.vertx.core.file.impl.FileResolver.CACHE_DIR_BASE_PROP_NAME;
 
 import java.io.File;
 import java.time.Duration;
-import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
@@ -13,8 +17,6 @@ import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
 import java.util.function.Supplier;
-import java.util.regex.Pattern;
-import java.util.stream.Collectors;
 
 import org.jboss.logging.Logger;
 import org.wildfly.common.cpu.ProcessorInfo;
@@ -36,18 +38,12 @@ import io.vertx.core.eventbus.EventBusOptions;
 import io.vertx.core.file.FileSystemOptions;
 import io.vertx.core.http.ClientAuth;
 import io.vertx.core.impl.VertxImpl;
-import io.vertx.core.net.JksOptions;
-import io.vertx.core.net.PemKeyCertOptions;
-import io.vertx.core.net.PemTrustOptions;
-import io.vertx.core.net.PfxOptions;
 import io.vertx.core.spi.resolver.ResolverProvider;
 
 @Recorder
 public class VertxCoreRecorder {
 
     private static final Logger LOGGER = Logger.getLogger(VertxCoreRecorder.class.getName());
-
-    private static final Pattern COMMA_PATTERN = Pattern.compile(",");
 
     static volatile VertxSupplier vertx;
 
@@ -257,54 +253,14 @@ public class VertxCoreRecorder {
         opts.setTrustAll(eb.trustAll);
 
         // Certificates and trust.
-        if (eb.keyCertificatePem != null) {
-            List<String> certs = new ArrayList<>();
-            List<String> keys = new ArrayList<>();
-            eb.keyCertificatePem.certs.ifPresent(
-                    s -> certs.addAll(COMMA_PATTERN.splitAsStream(s).map(String::trim).collect(Collectors.toList())));
-            eb.keyCertificatePem.keys.ifPresent(
-                    s -> keys.addAll(COMMA_PATTERN.splitAsStream(s).map(String::trim).collect(Collectors.toList())));
-            PemKeyCertOptions o = new PemKeyCertOptions()
-                    .setCertPaths(certs)
-                    .setKeyPaths(keys);
-            opts.setPemKeyCertOptions(o);
-        }
+        configurePemKeyCertOptions(opts, eb.keyCertificatePem);
+        configureJksKeyCertOptions(opts, eb.keyCertificateJks);
+        configurePfxKeyCertOptions(opts, eb.keyCertificatePfx);
 
-        if (eb.keyCertificateJks != null) {
-            JksOptions o = new JksOptions();
-            eb.keyCertificateJks.path.ifPresent(o::setPath);
-            eb.keyCertificateJks.password.ifPresent(o::setPassword);
-            opts.setKeyStoreOptions(o);
-        }
+        configurePemTrustOptions(opts, eb.trustCertificatePem);
+        configureJksKeyCertOptions(opts, eb.trustCertificateJks);
+        configurePfxTrustOptions(opts, eb.trustCertificatePfx);
 
-        if (eb.keyCertificatePfx != null) {
-            PfxOptions o = new PfxOptions();
-            eb.keyCertificatePfx.path.ifPresent(o::setPath);
-            eb.keyCertificatePfx.password.ifPresent(o::setPassword);
-            opts.setPfxKeyCertOptions(o);
-        }
-
-        if (eb.trustCertificatePem != null) {
-            eb.trustCertificatePem.certs.ifPresent(s -> {
-                PemTrustOptions o = new PemTrustOptions();
-                COMMA_PATTERN.splitAsStream(s).map(String::trim).forEach(o::addCertPath);
-                opts.setPemTrustOptions(o);
-            });
-        }
-
-        if (eb.trustCertificateJks != null) {
-            JksOptions o = new JksOptions();
-            eb.trustCertificateJks.path.ifPresent(o::setPath);
-            eb.trustCertificateJks.password.ifPresent(o::setPassword);
-            opts.setTrustStoreOptions(o);
-        }
-
-        if (eb.trustCertificatePfx != null) {
-            PfxOptions o = new PfxOptions();
-            eb.trustCertificatePfx.path.ifPresent(o::setPath);
-            eb.trustCertificatePfx.password.ifPresent(o::setPassword);
-            opts.setPfxTrustOptions(o);
-        }
         options.setEventBusOptions(opts);
     }
 

--- a/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/config/JksConfiguration.java
+++ b/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/config/JksConfiguration.java
@@ -9,6 +9,12 @@ import io.quarkus.runtime.annotations.ConfigItem;
 public class JksConfiguration {
 
     /**
+     * JKS config is disabled by default.
+     */
+    @ConfigItem(name = ConfigItem.PARENT, defaultValue = "false")
+    public boolean enabled;
+
+    /**
      * Path of the key file (JKS format).
      */
     @ConfigItem

--- a/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/config/PemKeyCertConfiguration.java
+++ b/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/config/PemKeyCertConfiguration.java
@@ -1,5 +1,6 @@
 package io.quarkus.vertx.core.runtime.config;
 
+import java.util.List;
 import java.util.Optional;
 
 import io.quarkus.runtime.annotations.ConfigGroup;
@@ -18,12 +19,12 @@ public class PemKeyCertConfiguration {
      * Comma-separated list of the path to the key files (Pem format).
      */
     @ConfigItem
-    public Optional<String> keys;
+    public Optional<List<String>> keys;
 
     /**
      * Comma-separated list of the path to the certificate files (Pem format).
      */
     @ConfigItem
-    public Optional<String> certs;
+    public Optional<List<String>> certs;
 
 }

--- a/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/config/PemKeyCertConfiguration.java
+++ b/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/config/PemKeyCertConfiguration.java
@@ -9,6 +9,12 @@ import io.quarkus.runtime.annotations.ConfigItem;
 public class PemKeyCertConfiguration {
 
     /**
+     * PEM Key/cert config is disabled by default.
+     */
+    @ConfigItem(name = ConfigItem.PARENT, defaultValue = "false")
+    public boolean enabled;
+
+    /**
      * Comma-separated list of the path to the key files (Pem format).
      */
     @ConfigItem

--- a/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/config/PemTrustCertConfiguration.java
+++ b/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/config/PemTrustCertConfiguration.java
@@ -9,6 +9,12 @@ import io.quarkus.runtime.annotations.ConfigItem;
 public class PemTrustCertConfiguration {
 
     /**
+     * PEM Trust config is disabled by default.
+     */
+    @ConfigItem(name = ConfigItem.PARENT, defaultValue = "false")
+    public boolean enabled;
+
+    /**
      * Comma-separated list of the trust certificate files (Pem format).
      */
     @ConfigItem

--- a/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/config/PemTrustCertConfiguration.java
+++ b/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/config/PemTrustCertConfiguration.java
@@ -1,5 +1,6 @@
 package io.quarkus.vertx.core.runtime.config;
 
+import java.util.List;
 import java.util.Optional;
 
 import io.quarkus.runtime.annotations.ConfigGroup;
@@ -18,6 +19,6 @@ public class PemTrustCertConfiguration {
      * Comma-separated list of the trust certificate files (Pem format).
      */
     @ConfigItem
-    public Optional<String> certs;
+    public Optional<List<String>> certs;
 
 }

--- a/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/config/PfxConfiguration.java
+++ b/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/config/PfxConfiguration.java
@@ -9,7 +9,13 @@ import io.quarkus.runtime.annotations.ConfigItem;
 public class PfxConfiguration {
 
     /**
-     * Path to the key file (PFX format)
+     * PFX config is disabled by default.
+     */
+    @ConfigItem(name = ConfigItem.PARENT, defaultValue = "false")
+    public boolean enabled;
+
+    /**
+     * Path to the key file (PFX format).
      */
     @ConfigItem
     public Optional<String> path;

